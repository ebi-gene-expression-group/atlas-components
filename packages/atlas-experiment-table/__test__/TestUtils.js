import React from 'react'
import styled from 'styled-components'

// Stolen from https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/random
const getRandomInt = (min, max) => {
  min = Math.ceil(min)
  max = Math.floor(max)
  return Math.floor(Math.random() * (max - min)) + min //The maximum is exclusive and the minimum is inclusive
}

const TableCellDiv = styled.div`
  font-size: 13px;
  font-family: Helvetica, Arial, FreeSans, "Liberation Sans", sans-serif;
`

const tableHeader = [
  { type: `sort`, title: `Loaded date`, width: 140, dataParam: `lastUpdate` },
  { type: `search`, title: `species`, width: 200, dataParam: `species` },
  { type: ``, title: `experiment description`, width: 360, dataParam: `experimentDescription`, link: `experimentAccession`, resource: `experiments`, endpoint: `Results` },
  { type: `plain`, title: `experiment factors`, width: 260, dataParam: `experimentalFactors` },
  { type: `sort`, title: `Number of assays`, width: 160, dataParam: `numberOfAssays`, link: `experimentAccession`, resource: `experiments`, endpoint: `Experiment Design` },
]

const data = [
  {
    experimentType: `SINGLE`,
    experimentAccession: `E-EHCA-2`,
    experimentDescription: `Melanoma infiltration`,
    lastUpdate: `16-11-2018`,
    numberOfAssays: 6638,
    numberOfContrasts: 0,
    species: `Mus musculus`,
    kingdom: `animals`,
    experimentalFactors: [`single cell identifier`, `sampling site`,`time`],
  },
  {
    experimentType: `SINGLE`,
    experimentAccession: `E-GEOD-99058`,
    experimentDescription: `Single cell`,
    lastUpdate: `11-10-2018`,
    numberOfAssays: 250,
    numberOfContrasts: 0,
    species: `Mus musculus`,
    kingdom: `animals`,
    experimentalFactors: [`single cell identifier`]
  },
  {
    experimentType: `SINGLE`,
    experimentAccession: `E-MTAB-5061`,
    experimentDescription: `healthy individuals and type 2 diabetes patients`,
    lastUpdate: `11-10-2018`,
    numberOfAssays: 3514,
    numberOfContrasts: 0,
    species: `Homo sapiens`,
    kingdom: `plants`,
    experimentalFactors: [`single cell identifier`, `disease`],
  }
]

export {getRandomInt, TableCellDiv, tableHeader, data}
