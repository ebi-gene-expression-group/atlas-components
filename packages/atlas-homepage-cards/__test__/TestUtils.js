

// Stolen from https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/random
const getRandomInt = (min, max) => {
  min = Math.ceil(min)
  max = Math.floor(max)
  return Math.floor(Math.random() * (max - min)) + min //The maximum is exclusive and the minimum is inclusive
}

const dummyCards = () => {
  return [
    {
      iconType: `species`,
      iconSrc: `mus musculus`,
      iconDescription: `Mus musculus`,
      content: [
        {
          text: `4 experiments`,
        }
      ]
    },
    {
      iconType: `species`,
      iconSrc: `homo sapiens`,
      iconDescription: `Homo sapiens`,
      content: [
        {
          text: `2 experiments`,
        }
      ]
    },
    {
      iconType: `unknown`,
      iconSrc: `http://www.lingyun.com`,
      iconDescription: `Mus musculus`,
      content: [
        {
          text: `4 experiments`,
        }
      ]
    },
    {
      iconType: `experiments`,
      iconSrc: `http://www.zhao.com`,
      iconDescription: `Homo sapiens`,
      content: [
        {
          text: `2 experiments`,
        }
      ]
    },
    {
      iconType: `image`,
      iconSrc: `logo-goes-here`,
      iconDescription: `An image card`,
      content: [
        {
          text: `Go to experiments`,
        }
      ]
    },
    {
      iconType: `imagespecies`,
      iconSrc: `http://www.lingyun.com`,
      iconDescription: `Mus musculus`,
      content: [
        {
          text: `4 experiments`,
        }
      ]
    },
  ]
}

export {getRandomInt, dummyCards}