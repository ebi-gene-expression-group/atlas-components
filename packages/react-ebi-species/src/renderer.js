"use strict";

//*------------------------------------------------------------------*

var React = require('react');
var ReactDOM = require('react-dom');

//*------------------------------------------------------------------*

var Icon = require('./SpeciesIcon.jsx');

//*------------------------------------------------------------------*

module.exports = function(mountNode,species,colourOverride, colourPerKingdom) {
    ReactDOM.render(
        React.createElement(Icon,{
          species: species,
          colourOverride: colourOverride,
          colourPerKingdom: colourPerKingdom
        }), mountNode
    );
};
