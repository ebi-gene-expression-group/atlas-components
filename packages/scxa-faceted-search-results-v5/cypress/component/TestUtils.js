import facets from '../fixtures/vindicatorsResponse.json'
import organismParts from '../fixtures/organismPartsResponse.json'
import cellTypes from '../fixtures/cellTypesResponse.json'
import species from '../fixtures/speciesResponse.json'

import PropTypes from "prop-types";
import styled from "styled-components";
import React from "react";
import CheckboxFacetGroupsDefaultProps from "../../src/facetgroups/CheckboxFacetGroupsDefaultProps";
import MultiSelectDropdownFacetGroupsDefaultProps
  from "../../src/facetgroups/MultiSelectDropdownFacetGroupsDefaultProps";

const getPropsForCheckBoxGroupWithTooltip = () => {
  return Cypress._.shuffle(CheckboxFacetGroupsDefaultProps).pop()
}

const getPropsForMultiSelectDropdownGroupWithTooltip = () => {
  return Cypress._.shuffle(MultiSelectDropdownFacetGroupsDefaultProps).pop()
}

const getPropsForMarkerGeneFacet = () => {
  const markerGeneName = `Marker genes`;
  return CheckboxFacetGroupsDefaultProps.find(facet => facet.name === markerGeneName)
}

const getPropsWithoutTooltip = () => {
  let props = getPropsForCheckBoxGroupWithTooltip()
  props.description = ``
  return props
}

const getFacets = () => {
  return Cypress._.shuffle(facets).filter(() => Math.random() > 0.5)
}

const getRandomSpecies = () => {
  return Cypress._.shuffle(species).filter(() => Math.random() > 0.5)
}

const getRandomOrganismParts = () => {
  return Cypress._.shuffle(organismParts).filter(() => Math.random() > 0.5)
}

const getRandomCellTypes = () => {
  return Cypress._.shuffle(cellTypes).filter(() => Math.random() > 0.5)
}

const ExperimentTableCard = ({title}) =>
  <div>
    <p>{title}</p>
  </div>

ExperimentTableCard.propTypes = {
  title: PropTypes.string.isRequired
}

const IconDiv = styled.div`
  width: 15%;
  text-align: center;
`
IconDiv.displayName = 'IconDiv'

const BitDiv = styled.div`
  width: 25%;
  text-align: center;
`
const DoveDiv = styled.div`
  width: 25%;
  text-align: center;
`

const ExperimentTableHeader = ({onClick}) =>
  <DoveDiv>
    <IconDiv onClick={()=>onClick(`test`)}>
      Meow
    </IconDiv>
    <BitDiv onClick={()=>onClick(`test`)}>
      Wow wow
    </BitDiv>
  </DoveDiv>

ExperimentTableHeader.propTypes = {
  onClick: PropTypes.func.isRequired
}

// Stolen from https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/random
const getRandomInt = (min, max) => {
  min = Math.ceil(min)
  max = Math.floor(max)
  return Math.floor(Math.random() * (max - min)) + min //The maximum is exclusive and the minimum is inclusive
}

export { getRandomInt, getFacets, getRandomOrganismParts, getRandomCellTypes, getRandomSpecies,
  getPropsForCheckBoxGroupWithTooltip, getPropsForMultiSelectDropdownGroupWithTooltip,
  getPropsWithoutTooltip, getPropsForMarkerGeneFacet,
  ExperimentTableCard, ExperimentTableHeader }
