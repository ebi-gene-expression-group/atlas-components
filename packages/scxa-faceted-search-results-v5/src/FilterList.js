import React from 'react'
import PropTypes from 'prop-types'
import _ from 'lodash'
class FilterList extends React.Component {
  constructor(props) {
    super(props)

    this.state = {
      ascending: false,
      sortTitle: props.sortTitle
    }

    this.sortEntries = this.sortEntries.bind(this)
  }
  sortEntries(attribute){
    this.setState({
      sortTitle: attribute,
      ascending: !this.state.ascending
    })
  }

  render() {
    const {filteredResults, resultsMessage, ResultsHeaderClass, ResultElementClass} = this.props

    const filteredElements = _.sortBy(filteredResults, this.state.sortTitle)
    const sortedElements = this.state.ascending ? filteredElements :  filteredElements.reverse()

    return (
      <div id={`filterList`}>
        <h4>{resultsMessage}</h4>
        { sortedElements.length && <ResultsHeaderClass onClick={this.sortEntries}/> }
        { sortedElements.map((element, index) => <div key={index}><ResultElementClass {...element}/></div>) }
      </div>
    )
  }
}

FilterList.propTypes = {
  filteredResults: PropTypes.arrayOf(PropTypes.object).isRequired,
  resultsMessage: PropTypes.string,
  ResultsHeaderClass: PropTypes.func,
  ResultElementClass: PropTypes.func.isRequired,
  sortTitle: PropTypes.string
}

FilterList.defaultProps = {
  resultsMessage: ``
}

export default FilterList
