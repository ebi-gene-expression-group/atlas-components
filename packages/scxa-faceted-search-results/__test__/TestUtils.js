import React from 'react'
import PropTypes from 'prop-types'

const EpisodeCard = ({title}) =>
  <div>
    <p>{title}</p>
  </div>

EpisodeCard.propTypes = {
  title: PropTypes.string.isRequired
}

// Stolen from https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/random
const getRandomInt = (min, max) => {
  min = Math.ceil(min);
  max = Math.floor(max);
  return Math.floor(Math.random() * (max - min)) + min; //The maximum is exclusive and the minimum is inclusive
}

const vindicators = [
  { group:`Vindicator`, value: `supernova`, label: `Supernova`},
  { group:`Vindicator`, value: `noob-noob`, label: `Noob-Noob` },
  { group:`Vindicator`, value: `vance_maximus`, label: `Vance Maximus`},
  { group:`Vindicator`, value: `alan_rails`, label: `Alan Rails`},
  { group:`Vindicator`, value: `crocubot`, label: `Crocubot` },
  { group:`Vindicator`, value: `million_ants`, label: `Million Ants`},
  { group:`Vindicator`, value: `morty_smith`, label: `Morty Smith`},
  { group:`Vindicator`, value: `rick_sanchez`, label: `Rick Sanchez`},
  { group:`Vindicator`, value: `lady_katana`, label: `Lady Katana`},
  { group:`Vindicator`, value: `calypso`, label: `Calypso`},
  { group:`Vindicator`, value: `diablo_verde`, label: `Diablo Verde`}
]

const episodes = [
  {
    element: {
      title: "Raising Gazorpazorp"
    },
    facets: [
      {
        group: "Planet",
        value: "gazorpazorp",
        label: "Gazorpazorp"
      },
      {
        group: "Guest character",
        value: "gwendolyn",
        label: "Gwendolyn"
      },
      {
        group: "Guest character",
        value: "ma-sha",
        label: "Ma-Sha"
      },
      {
        group: "Season",
        value: "1",
        label: "1"
      }
    ]
  },
  {
    element: {
      title: "The wedding squanchers"
    },
    facets: [
      {
        group: "Planet",
        value: "squanch",
        label: "Squanch"
      },
      {
        group: "Guest character",
        value: "birdperson",
        label: "Birdperson"
      },
      {
        group: "Guest character",
        value: "squanchy",
        label: "Squanchy"
      },
      {
        group: "Season",
        value: "2",
        label: "2"
      }
    ]
  },
  {
    element: {
      title: "The Rickshank redemption"
    },
    facets: [
      {
        group: "Guest character",
        value: "birdperson",
        label: "Birdperson"
      },
      {
        group: "Planet",
        value: "buttworld",
        label: "Buttworld"
      },
      {
        group: "Season",
        value: "3",
        label: "3"
      }
    ]
  },
  {
    element: {
      title: "Ricksy business"
    },
    facets: [
      {
        group: "Guest character",
        value: "squanchy",
        label: "Squanchy"
      },
      {
        group: "Guest character",
        value: "abradolf_lincler",
        label: "Abradolf Lincler"
      },
      {
        group: "Season",
        value: "1",
        label: "1"
      }
    ]
  },
  {
    element: {
      title: "Close Rick-counters of the Rick kind"
    },
    facets: [
      {
        group: "Guest character",
        value: "ricktiminus_sancheziminius",
        label: "Ricktiminus Sancheziminius"
      },
      {
        group: "Guest character",
        value: "abradolf_lincler",
        label: "Abradolf Lincler"
      },
      {
        group: "Planet",
        value: "buttworld",
        label: "Buttworld"
      },
      {
        group: "Season",
        value: "1",
        label: "1"
      }
    ]
  }
]



export {getRandomInt, vindicators, episodes, EpisodeCard}
