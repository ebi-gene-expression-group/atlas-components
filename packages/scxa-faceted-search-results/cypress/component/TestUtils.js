import episodes from '../fixtures/episodes.json'
import PropTypes from "prop-types";
import styled from "styled-components";
import React from "react";

const facets = () => {
  return episodes.reduce((acc, episode) => acc.concat(episode.facets), [])
}

const getUniqueFacets = () => {
  const allFacets = facets()
  return allFacets
    .filter((facet, index) => allFacets.findIndex((thatFacet) => facet.value === thatFacet.value) === index)
    .map((facet) => ({
      ...facet,
      disabled: false
    }))
}

const getFacetTooltip = () => {
  return Cypress._.shuffle(getUniqueFacets()).find(facet => facet.description)
}

const getFacetWithoutTooltip = () => {
  return Cypress._.shuffle(getUniqueFacets()).find(facet => !facet.description)
}

const getPropsWithTooltip = (facetTooltip) => {
  return {
    facetGroupName: facetTooltip.group,
      facetGroupNameDescription: facetTooltip.description,
    facets: getUniqueFacets().filter(facet => facet.group === facetTooltip.group),
    onChange: () => {}
  }
}

const getPropsWithoutTooltip = () => {
  const facetWithoutTooltip = getFacetWithoutTooltip()
  return {
    facetGroupName: facetWithoutTooltip.group,
      facets: getUniqueFacets().filter(facet => facet.group === facetWithoutTooltip.group),
    onChange: () => {}
  }
}

const ExperimentTableCard = ({title}) =>
  <div>
    <p>{title}</p>
  </div>

ExperimentTableCard.propTypes = {
  title: PropTypes.string.isRequired
}

const IconDiv = styled.div`
  width: 15%;
  text-align: center;
`
IconDiv.displayName = 'IconDiv'

const BitDiv = styled.div`
  width: 25%;
  text-align: center;
`
const DoveDiv = styled.div`
  width: 25%;
  text-align: center;
`

const ExperimentTableHeader = ({onClick}) =>
  <DoveDiv>
    <IconDiv onClick={()=>onClick(`test`)}>
      Meow
    </IconDiv>
    <BitDiv onClick={()=>onClick(`test`)}>
      Wow wow
    </BitDiv>
  </DoveDiv>

ExperimentTableHeader.propTypes = {
  onClick: PropTypes.func.isRequired
}

// Stolen from https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/random
const getRandomInt = (min, max) => {
  min = Math.ceil(min)
  max = Math.floor(max)
  return Math.floor(Math.random() * (max - min)) + min //The maximum is exclusive and the minimum is inclusive
}

export { episodes, getRandomInt, getUniqueFacets, getFacetTooltip, getPropsWithTooltip, getPropsWithoutTooltip,
  ExperimentTableCard, ExperimentTableHeader }
