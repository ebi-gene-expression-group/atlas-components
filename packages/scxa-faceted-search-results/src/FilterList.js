import React from 'react'
import PropTypes from 'prop-types'
import _ from 'lodash'

import FilterSidebar from './FilterSidebar'

class FilterList extends React.Component {
  constructor(props) {
    super(props)

    // TODO Build initial state of checked filters from props if wrapping in React Router to make it RESTful
    this.state = {
      selectedFacets: {}
    }

    this._handleChange = this._handleChange.bind(this)
  }

  _handleChange(facetGroupName, facets) {
    const nextSelectedFacets = _.defaultsDeep({}, this.state.selectedFacets)
    nextSelectedFacets[facetGroupName] = facets.map((facet) => facet.value)  // Store values only, discard labels

    this.setState({
      selectedFacets: nextSelectedFacets
    })
  }

  render() {
    const {results, checkboxFacetGroups, hideFacetGroupNames, ResultElementComponent, resultsMessage} = this.props
    const {selectedFacets} = this.state

    const filteredElements =
      results.filter((result) =>
        // Results that have at least one matching facet (some) in all (every) selected groups with at least one selected item
        Object.entries(selectedFacets)
          .filter(([selectedFacetName, selectedFacetValues]) => selectedFacetValues.length)
          .every(([selectedFacetName, selectedFacetValues]) =>
            result.facets.some((resultFacet) =>
              selectedFacetName === resultFacet.group && selectedFacetValues.includes(resultFacet.value))))
          .map((result) => result.element)

      const resultsHaveFacets =
        results.some((result) =>
          result.facets &&
          result.facets.some((facet) =>
            typeof(facet.group) === `string` && typeof(facet.value) === `string` && typeof(`label`) === `string` &&
            facet.group.length && facet.value.length && facet.value.length))

    return(
      <div className={`expanded row`}>
        { resultsHaveFacets &&
          <div className={`small-12 medium-2 columns`}>
            <FilterSidebar {...{checkboxFacetGroups, hideFacetGroupNames}}
                           facets={_(results).flatMap('facets').value()}
                           onChange={this._handleChange}/>
          </div>
        }
        <div className={resultsHaveFacets ? `small-12 medium-10 columns` : `small-12 columns`}>
          <h3>{resultsMessage}</h3>
            {filteredElements.map((element, index) => <div key={index}><ResultElementComponent {...element}/></div>)}
        </div>
      </div>
    )
  }
}

FilterList.propTypes = {
  // A list of results from where facets can be inferred by inspecting each result’s facets field
  // Supplied by FetchLoader, or set manually for testing
  results: PropTypes.arrayOf(PropTypes.shape({
    element: PropTypes.object.isRequired,
    facets: PropTypes.arrayOf(PropTypes.shape({
      group: PropTypes.string.isRequired,
      value: PropTypes.string.isRequired,
      label: PropTypes.string.isRequired
    }))
  })).isRequired,
  checkboxFacetGroups: PropTypes.arrayOf(PropTypes.string),
  hideFacetGroupNames: PropTypes.arrayOf(PropTypes.string),
  ResultElementComponent: PropTypes.func.isRequired, // must be a React.Component, sadly there’s no such prop :(
  resultsMessage: PropTypes.string
}

FilterList.defaultProps = {
  checkboxFacetGroups: [],
  hideFacetGroupNames: []
}

export default FilterList
