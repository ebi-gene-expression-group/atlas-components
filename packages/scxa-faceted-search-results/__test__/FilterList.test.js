import React from 'react'
import renderer from 'react-test-renderer'
import Enzyme from 'enzyme'
import { mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import { episodes, ExperimentTableHeader, ExperimentTableCard } from './TestUtils'

import FilterList from '../src/FilterList'

Enzyme.configure({ adapter: new Adapter() })

const props = {
  filteredResults: episodes,
  ResultsHeaderClass: ExperimentTableHeader,
  ResultElementClass: ExperimentTableCard
}

describe(`FilterList`, () => {
  test(`renders a table header and as many components of ResultElementClass as filtered results`, () => {
    const wrapper = mount(<FilterList {...props} />)
    expect(wrapper.find(ExperimentTableHeader)).toHaveLength(1)
    expect(wrapper.find(ExperimentTableCard)).toHaveLength(episodes.length)
  })

  test(`sorts table contents by clicking/toggling on headers`, () => {
    const wrapper = mount(<FilterList {...props} />)
    const sortOrder = wrapper.state(`ascending`)
    wrapper.find(`IconDiv`).simulate(`click`)
	  expect(wrapper.state(`ascending`)).toBe(!sortOrder)
	  wrapper.find(`IconDiv`).simulate(`click`)
	  expect(wrapper.state(`ascending`)).toBe(sortOrder)
  })

  test(`matches snapshot`, () => {
    const tree = renderer.create(<FilterList {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
