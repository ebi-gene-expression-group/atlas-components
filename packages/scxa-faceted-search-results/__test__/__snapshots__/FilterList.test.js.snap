// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterList matches snapshot 1`] = `
<div
  className="row expanded"
>
  <div
    className="small-12 medium-2 columns"
  >
    <div
      className="padding-bottom-xlarge"
    >
      <h4>
        Season
      </h4>
      <div>
        <input
          checked={false}
          onChange={[Function]}
          type="checkbox"
          value="1"
        />
        <label>
          Season 1
        </label>
      </div>
      <div>
        <input
          checked={false}
          onChange={[Function]}
          type="checkbox"
          value="2"
        />
        <label>
          Season 2
        </label>
      </div>
      <div>
        <input
          checked={false}
          onChange={[Function]}
          type="checkbox"
          value="3"
        />
        <label>
          Season 3
        </label>
      </div>
    </div>
    <div
      className="input-clear padding-bottom-xlarge"
    >
      <h4>
        Planet
      </h4>
      <div
        className="css-10nd86i "
        id={undefined}
        onKeyDown={[Function]}
      >
        <span
          aria-atomic="true"
          aria-live="polite"
          className="css-1uvvha"
          role="status"
        >
          3 results available.
        </span>
        <div
          className="css-12emboz "
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
        >
          <div
            className="css-nbc25y "
          >
            <div
              className="css-1492t68 "
            >
              Select...
            </div>
            <div
              className="css-rsyb7x"
            >
              <div
                className=""
                style={
                  Object {
                    "display": "inline-block",
                  }
                }
              >
                <input
                  aria-activedescendant={undefined}
                  aria-autocomplete="list"
                  aria-busy={false}
                  aria-describedby={undefined}
                  aria-expanded={false}
                  aria-haspopup={false}
                  aria-label={undefined}
                  aria-labelledby={undefined}
                  aria-owns={undefined}
                  autoCapitalize="none"
                  autoComplete="off"
                  autoCorrect="off"
                  className={undefined}
                  disabled={false}
                  id="react-select-10-input"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  role="combobox"
                  spellCheck="false"
                  style={
                    Object {
                      "background": 0,
                      "border": 0,
                      "boxSizing": "content-box",
                      "color": "inherit",
                      "fontSize": "inherit",
                      "opacity": 1,
                      "outline": 0,
                      "padding": 0,
                      "width": "1px",
                    }
                  }
                  tabIndex="0"
                  type="text"
                  value=""
                />
                <div
                  style={
                    Object {
                      "height": 0,
                      "left": 0,
                      "overflow": "scroll",
                      "position": "absolute",
                      "top": 0,
                      "visibility": "hidden",
                      "whiteSpace": "pre",
                    }
                  }
                >
                  
                </div>
              </div>
            </div>
          </div>
          <div
            className="css-1wy0on6 "
          >
            <span
              style={
                Object {
                  "backgroundImage": "url(\\"data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' version='1.1' width='32' height='24' viewBox='0 0 32 24'><polygon points='0,0 32,0 16,24' style='fill: rgb%28138, 138, 138%29'></polygon></svg>\\")",
                  "backgroundOrigin": "content-box",
                  "backgroundPosition": "center",
                  "backgroundRepeat": "no-repeat",
                  "backgroundSize": "9px 6px",
                  "height": "1rem",
                  "width": "1.5rem",
                }
              }
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="input-clear padding-bottom-xlarge"
    >
      <h4>
        Guest character
      </h4>
      <div
        className="css-10nd86i "
        id={undefined}
        onKeyDown={[Function]}
      >
        <span
          aria-atomic="true"
          aria-live="polite"
          className="css-1uvvha"
          role="status"
        >
          6 results available.
        </span>
        <div
          className="css-12emboz "
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
        >
          <div
            className="css-nbc25y "
          >
            <div
              className="css-1492t68 "
            >
              Select...
            </div>
            <div
              className="css-rsyb7x"
            >
              <div
                className=""
                style={
                  Object {
                    "display": "inline-block",
                  }
                }
              >
                <input
                  aria-activedescendant={undefined}
                  aria-autocomplete="list"
                  aria-busy={false}
                  aria-describedby={undefined}
                  aria-expanded={false}
                  aria-haspopup={false}
                  aria-label={undefined}
                  aria-labelledby={undefined}
                  aria-owns={undefined}
                  autoCapitalize="none"
                  autoComplete="off"
                  autoCorrect="off"
                  className={undefined}
                  disabled={false}
                  id="react-select-11-input"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  role="combobox"
                  spellCheck="false"
                  style={
                    Object {
                      "background": 0,
                      "border": 0,
                      "boxSizing": "content-box",
                      "color": "inherit",
                      "fontSize": "inherit",
                      "opacity": 1,
                      "outline": 0,
                      "padding": 0,
                      "width": "1px",
                    }
                  }
                  tabIndex="0"
                  type="text"
                  value=""
                />
                <div
                  style={
                    Object {
                      "height": 0,
                      "left": 0,
                      "overflow": "scroll",
                      "position": "absolute",
                      "top": 0,
                      "visibility": "hidden",
                      "whiteSpace": "pre",
                    }
                  }
                >
                  
                </div>
              </div>
            </div>
          </div>
          <div
            className="css-1wy0on6 "
          >
            <span
              style={
                Object {
                  "backgroundImage": "url(\\"data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' version='1.1' width='32' height='24' viewBox='0 0 32 24'><polygon points='0,0 32,0 16,24' style='fill: rgb%28138, 138, 138%29'></polygon></svg>\\")",
                  "backgroundOrigin": "content-box",
                  "backgroundPosition": "center",
                  "backgroundRepeat": "no-repeat",
                  "backgroundSize": "9px 6px",
                  "height": "1rem",
                  "width": "1.5rem",
                }
              }
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="small-12 medium-10 columns"
  >
    <h4 />
    <div>
      <div>
        <p>
          Raising Gazorpazorp
        </p>
      </div>
    </div>
    <div>
      <div>
        <p>
          The wedding squanchers
        </p>
      </div>
    </div>
    <div>
      <div>
        <p>
          The Rickshank redemption
        </p>
      </div>
    </div>
    <div>
      <div>
        <p>
          Ricksy business
        </p>
      </div>
    </div>
    <div>
      <div>
        <p>
          Close Rick-counters of the Rick kind
        </p>
      </div>
    </div>
  </div>
</div>
`;
